heroic-component/src/main/java/com/spotify/heroic/metric/MetricsConnectionSettings.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2015 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.heroic.metric;
+
+import static com.spotify.heroic.metric.consts.ApiQueryConsts.DEFAULT_MAX_ELAPSED_BACKOFF_MILLIS;
+import static com.spotify.heroic.metric.consts.ApiQueryConsts.DEFAULT_MAX_SCAN_TIMEOUT_RETRIES;
+import static com.spotify.heroic.metric.consts.ApiQueryConsts.DEFAULT_MUTATE_RPC_TIMEOUT_MS;
+import static com.spotify.heroic.metric.consts.ApiQueryConsts.DEFAULT_READ_ROWS_RPC_TIMEOUT_MS;
+import static com.spotify.heroic.metric.consts.ApiQueryConsts.DEFAULT_SHORT_RPC_TIMEOUT_MS;
+
+import com.spotify.heroic.metric.consts.ApiQueryConsts;
+import java.util.Optional;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+public class MetricsConnectionSettings {
+    /* default number of Cells for each batch mutation */
+    public static final int DEFAULT_MUTATION_BATCH_SIZE = 1_000;
+
+    /* maximum possible number of Cells for each batch mutation */
+    public static final int MAX_MUTATION_BATCH_SIZE = 100_000;
+
+    /* minimum possible number of Cells supported for each batch mutation */
+    public static final int MIN_MUTATION_BATCH_SIZE = 10;
+
+    /**
+     * See {@link com.spotify.heroic.metric.consts.ApiQueryConsts#DEFAULT_MUTATE_RPC_TIMEOUT_MS}
+     */
+    protected Integer mutateRpcTimeoutMs;
```

Comment:
I believe all of these should be non-nullable primitive `int` - they were originally and they all have defaults applied in the constructor.